lib/packages_file.dart

```diff
@@ -106,10 +106,15 @@ Map<String, Uri> parse(List<int> source, Uri baseLocation,
 /// If [baseUri] is provided, package locations will be made relative
 /// to the base URI, if possible, before writing.
 ///
+/// If [allowDefaultPackage] is `true`, the [packageMapping] may contain an
+/// empty string mapping to the _default package name_.
+///
 /// All the keys of [packageMapping] must be valid package names,
 /// and the values must be URIs that do not have the `package:` scheme.
 void write(StringSink output, Map<String, Uri> packageMapping,
-    {Uri baseUri, String comment}) {
+    {Uri baseUri, String comment, bool allowDefaultPackage = false}) {
+  ArgumentError.checkNotNull(allowDefaultPackage, 'allowDefaultPackage');
```

Comment:
I wouldn't bother with this check. It will fail below if the value is `null`, and an optional parameter with a non-`null` default value is *only* `null` if someone explicitly passes that. 
It is not documented that `null` is accepted, so that should be assumed to be an error, and so rare that I don't want to add code for it.

(When comes NNBD, this check will be unnecessary anyway).

Now it's here, it's fine, but not something I'd usually bother doing.